File: javascript/07_maps.js
// 07 objects / maps

const person = {
  name: "ulas",
  age: 20,
  city: "istanbul"
};

console.log(person.name);
console.log(person["age"]);

person.age = 21;
person.job = "student";
console.log(person);

for (const [key, value] of Object.entries(person)) {
  console.log(key, "->", value);
}

File: cpp/07_maps.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

int main() {
    map<string, string> person;
    person["name"] = "ulas";
    person["city"] = "istanbul";

    cout << person["name"] << endl;

    person["job"] = "student";

    for (auto const& [key, value] : person) {
        cout << key << " -> " << value << endl;
    }

    return 0;
}
